app/collections/favorites-button.cjsx

```diff
@@ -0,0 +1,89 @@
+React = require 'react'
+apiClient = require '../api/client'
+auth = require '../api/auth'
+getFavoritesName = require './get-favorites-name'
+alert = require '../lib/alert'
+
+module?.exports = React.createClass
+  displayName: 'CollectionFavoritesButton'
+
+  propTypes:
+    subject: React.PropTypes.object # a subject response from panoptes
+
+  getInitialState: ->
+    favorited: false
+
+  componentWillMount: ->
+    project_id = @props.subject.links.project.toString()
```

Comment:
I wanna say almost never access the `links` field directly, if you can help it. You can request a subject's project with `subject.get('project').then (project) ->`.